var/spack/repos/builtin/packages/legion/package.py

```diff
@@ -147,10 +146,19 @@ def validate_gasnet_root(value):
             values=cuda_arch_list,
             description="GPU/CUDA architecture to build for.",
             multi=False)
+    conflicts('+cuda_arch', when='~cuda')
```

Comment:
`cuda_arch` is not a boolean variant, so `+cuda_arch` won't apply. If you extend CudaPackage the conflicts are set up for you already! https://github.com/spack/spack/blob/develop/lib/spack/spack/build_systems/cuda.py#L185-L187

Reply:
Thanks.  Definitely, on my "todo" list but just not enough cycles right now to get all the changes in.